mesonbuild/interpreter.py

```diff
@@ -3761,7 +3761,7 @@ def unpack_env_kwarg(self, kwargs) -> build.EnvironmentVariables:
 
     def add_test(self, node, args, kwargs, is_base_test):
         if len(args) != 2:
-            raise InterpreterException('Incorrect number of arguments')
+            raise InterpreterException(r'test expects 2 arguments, only {len(args)} given')
```

Comment:
```suggestion
            raise InterpreterException(r'test expects 2 arguments, {len(args)} given')
```
"test expects 2 arguments, only 3 given" doesn't really make that much sense.